src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURE_COUNT = 5;
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int figureRandom = random.nextInt(FIGURE_COUNT);
+        switch (figureRandom) {
+            case 0:
+                return new Circle(random.nextInt(40));
+            case 1:
+                return new IsoscelesTrapezoid(random.nextInt(40),
+                        random.nextInt(40), random.nextInt(40));
+            case 2:
+                return new Rectangle(random.nextInt(40), random.nextInt(40));
+            case 3:
+                return new RightTriangle(random.nextInt(40), random.nextInt(40));
+            default:
+                return new Square(random.nextInt(40));
```

Comment:
don't pass expressions in constructor arguments. let's create separate variables and pass them.  and to keep code clean do all of these in a private method.